solana/solana-ibc/programs/solana-ibc/src/execution_context.rs

```diff
@@ -238,17 +240,31 @@ impl ExecutionContext for IbcStorage<'_, '_> {
         commitment: PacketCommitment,
     ) -> Result {
         msg!("store_packet_commitment({path}, {commitment:?})");
+        let mut store = self.borrow_mut();
         let trie_key = TrieKey::from(path);
         // PacketCommitment is always 32-byte long.
         let commitment = <&CryptoHash>::try_from(commitment.as_ref()).unwrap();
-        self.borrow_mut().provable.set(&trie_key, commitment).unwrap();
+        store.provable.set(&trie_key, commitment).unwrap();
+        record_packet_sequence(
```

Comment:
This could fail right?

Reply:
The record_packet_sequence call? No. It just adds value to a vector.

Reply:
But don't we have limits on space?

Reply:
Yes, in this sense it can fail indeed. We’ve been discussing this quite a bit actually and the current thinking is that this should be fine. Commitments are removed when packets are acknowledged which means that size of this list is equal to number of unacknowledged packets. Similarly, commitments are removed from the trie so its size should be fine as well.

The bigger issue is acknowledgement commitments which are never deleted. However, we believe that we can simply ignore acks and not store them at all.

Lastly, there are also receipts but those are held as sealed in the trie so they should not increase storage size only.

Long story short, for the time being this should be fine but we will definitely come back to it.